analyzers/tests/SonarAnalyzer.Test/TestCases/RouteTemplateShouldNotStartWithSlash.AspNet4x.cs

```diff
@@ -0,0 +1,136 @@
+
+    [Route("/SubPath/Index4_1")]        // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [Route("/[controller]/Index4_2")]   // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    public ActionResult Index4() => View();
+}
+
+[RoutePrefix("[controller]")]
+public class NoncompliantWithRoutePrefixController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.}}
+//           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+{
+    [Route("/Index1")]                  // Secondary
+//   ^^^^^^^^^^^^^^^^
+    public ActionResult Index1() => View();
+}
+
+[Route("[controller]")]
+[Route("[controller]/[action]")]
+public class NoncompliantMultiRouteController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.}}
+//           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+{
+    [Route("/Index1")]                  // Secondary
+//   ^^^^^^^^^^^^^^^^
+    public ActionResult Index1() => View();
+
+    [Route("/SubPath/Index2_1")]        // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [Route("/[controller]/Index2_2")]   // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    public ActionResult Index2() => View();
+
+    [Route("/[action]")]                // Secondary
+//   ^^^^^^^^^^^^^^^^^^
+    public ActionResult Index3() => View();
+}
+
+[Route("[controller]")]
+public class CompliantController : Controller // Compliant: at least one action has at least a relative route
+{
+    [Route("/Index1")]
+    public ActionResult Index1() => View();
+
+    [Route("/SubPath/Index2")]
+    public ActionResult Index2() => View();
+
+    [Route("/[action]")]
+    public ActionResult Index3() => View();
+
+    [Route("/[controller]/Index4_1")]
+    [Route("SubPath/Index4_2")] // The relative route
+    public ActionResult Index4() => View();
+}
+
+public class NoncompliantNoControllerRouteController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and add a controller route with the common prefix.}}
+{
+    [Route("/Index1")]                          // Secondary
+    public ActionResult Index1() => View();
+
+    [Route("/SubPath/Index2_1")]                // Secondary
+    [Route("/[controller]/Index2_2")]           // Secondary
+    public ActionResult Index2() => View();
+
+    [Route("/[action]")]                        // Secondary
+    public ActionResult Index3() => View();
+}
+
+public class CompliantNoControllerRouteNoActionRouteController : Controller // Compliant
+{
+    public ActionResult Index1() => View(); // Default route -> relative
+
+    [Route("/SubPath/Index2")]
+    public ActionResult Index2() => View();
+
+    [Route("/[action]")]
+    public ActionResult Index3() => View();
+
+    [Route("/SubPath/Index4_1")]
+    [Route("/[controller]/Index4_2")]
+    public ActionResult Index4() => View();
+}
+
```

Comment:
Add some tests where other route properties are set to "/"
```cs
public class MyController : Controller // Compliant
{
    [Route("compliant", Name = "/Valid")] // Slash in "Name" is compliant
    public ActionResult Index1() => View(); 
}
```

Reply:
Will do in the parametrized tests PR.
https://github.com/SonarSource/rspec/blob/d9e454789cdfbdbb19c8266a106bfa32ed85f825/rules/S6931/S6931.AspNet4x.cs#L147